phoenix-core-server/src/main/java/org/apache/phoenix/hbase/index/IndexRegionObserver.java

```diff
@@ -625,7 +639,7 @@ private Collection<? extends Mutation> groupMutations(MiniBatchOperationInProgre
     public static void setTimestamps(MiniBatchOperationInProgress<Mutation> miniBatchOp,
                                      IndexBuildManager builder, long ts) throws IOException {
         for (Integer i = 0; i < miniBatchOp.size(); i++) {
-            if (miniBatchOp.getOperationStatus(i) == IGNORE) {
+            if (miniBatchOp.getOperationStatus(i).getResult() != null) {
```

Comment:
Same here. Will the IGNORE check won't work ?

Reply:
We no longer rely on `IGNORE` because this code has been removed:

```
                  // empty list of generated mutations implies ON DUPLICATE KEY IGNORE
                  miniBatchOp.setOperationStatus(i, IGNORE);
```

Reply:
We can't remove it. Otherwise how will ON DUPLICATE KEY IGNORE work ? You have to mark it IGNORED.

Reply:
How is then hbase ignoring the mutation because it is still there in the batch ?

Reply:
IGNORE and SUCCESS are equivalent so I don't understand what's the difference.

Reply:
FYI https://github.com/apache/phoenix/pull/1884#discussion_r1618041568

> Here, HBase sets OperationStatus.SUCCESS to batchOp.retCodeDetails[i] for Put/Delete mutations, leaving no Result object around with OperationStatus.

Reply:
What does it mean for the result to be null vs non null ? It is not very intuitive  when reading the code. We can introduce a utility function here something like this:
 `isAtomicOperationComplete(OperationStatus status) {
return status.getOperationStatusCode() == SUCCESS && status.getResult() != null
}`

Reply:
Creating utility function is fine, which can be used in all similar places.

Reply:
Though even then, any dev would still not get clarity on why we differentiate null vs non null, so as part of the javadoc of the new utility method, we should mention specifically that 
"HBase returns null Result by default for successful operation, only Increment and Append return non-null Result."

This will make things clear for whoever reads this code in future.

Reply:
Done